### ISSUE TITLE
cache/memory fills up over the course of a week.

### ISSUE THREAD

**Issue by giffmann** [mattermost/mattermost-mobile#7322]
Per Mattermost guidelines, GitHub issues are for bug reports: <http://www.mattermost.org/filing-issues/>.

For troubleshooting see: http://forum.mattermost.org/.
For feature proposals see: http://www.mattermost.org/feature-requests/

If you've found a bug--something appears unintentional--please follow these steps:

1. Confirm you’re filing a new issue. [Search existing tickets in Jira](https://mattermost.atlassian.net/jira/software/c/projects/MM/issues/) to ensure that the ticket does not already exist.
2. Confirm your issue does not involve security. Otherwise, please see our [Responsible Disclosure Policy](https://about.mattermost.com/report-security-issue/).
3. [File a new issue](https://github.com/mattermost/mattermost-mobile/issues/new) using the format below. Mattermost will confirm steps to reproduce and file in Jira, or ask for more details if there is trouble reproducing it. If there's already an existing bug in Jira, it will be linked back to the GitHub issue so you can track when it gets fixed.

#### Summary
On a fresh install of the iOS app (v2.3.0) the app will work good and be responsive for about a week.  At around the 1 week timeframe, the app starts becoming very slow to respond to any actions.  A clean install fixes the issue but it again rears it's head in about a week.

### Environment Information
  - Device Name: iPhone 13
  - OS Version: 16.4.1
  - Mattermost App Version: 2.3.0
  - Mattermost Server Version: 7.8.2

#### Steps to reproduce
Use the app a LOT with 3 teams.  Watch performance degrade to the point where sends/loads/touch actions take 8 to 10 seconds long to perform

#### Expected behavior
SLOW response to ANY action

#### Observed behavior (that appears unintentional)
Same as above

#### Possible fixes
Seems like a memory leak or cache is filling and unable to empty.
A fresh install fixes the issues.

**Comment by profyan99** [comment]
We have encountered the same behavior. It seems that you need to periodically clean the SQLite database, which is gradually filled with data

**Comment by giffmann** [comment]
Updated my original post to clean it up some.

I should also note I have multiple members of my team that report the same issue.

A bit surprised there has been zero response from Mattermost considering the issue has been open over a month.

**Comment by runlevel-six** [comment]
I can confirm that I have to uninstall/reinstall the Mattermost mobile app on iOS 16 about once a week - by the 6 or 7th day it is running so slow as to be unusable - for example 10 seconds to swap channels.

**Comment by giffmann** [comment]
Would be really nice to see this resolved.  Believe it’s related to the #7202 as well

getting real ridiculous to have to uninstall/install weekly to use the app

**Comment by radokristof** [comment]
@giffmann I'm sure that it is related, basically the same issue.
I just tried reinstalling the app now and it solved the slowness... I will watch how many days this can last.

**Comment by hlesesne** [comment]
Bump.

**Comment by giffmann** [comment]
Getting worse and worse.  Hard to believe a bug like this has not been addressed in nearly a year.  

Now the app just randomly closes as well.  Reinstalls do not help.  I'm now having to reinstall the app every 2 days or so!  Totally ridiculous.  

Please use some of that Air Force money to actually fix this app!!!!!

**Comment by giffmann** [comment]
FYI:  This affects, easily, close to 200 members of my IT Organization.

**Comment by hlesesne** [comment]
Bump. I don't understand how this issue hasn't gotten more attention. I've resorted to using mattermost mobile just for notifications, and using the the web version from safari on iOS. For a while it was just getting slow, and a reinstall would fix it for a few days. At this point, the entire app unloads on iOS after a few seconds (sometimes minutes).

**Comment by WStringer2** [comment]
Ditto - it's unusable with only 2 teams.

**Comment by giffmann** [comment]
@amyblais This is the other issue I filed that does not seem to have ever been looked at.  Same basic issue as the one I opened this morning.  Both issues were asked for by whoever monitors the iOS app store ratings.  Would be great to finally get some attention after nearly a year.

**Comment by amyblais** [comment]
Opened https://mattermost.atlassian.net/browse/MM-57322.

**Comment by giffmann** [comment]
> Opened https://mattermost.atlassian.net/browse/MM-57322.

Thanks you

**Comment by giffmann** [comment]
At this point, the app is totally useless for any length of time.  This has a very serious impact on my workflow.  I'm sure this affects MANY other users as well.

**Comment by giffmann** [comment]
@amyblais what does it take to make this a higher priority.  The app is all but useless at this point for a very large number of our users and this experience has only been degrading over the last year.
Thanks!

**Comment by giffmann** [comment]
@amyblais any updates?  It really is annoying to have to delete the app and reinstall every few days?  I don't see any updates in the referenced Jira ticket in over 60 days.  I know this is affecting many folks in my organization, and I would assume many other users.
Thanks!

**Comment by fastndead** [comment]
> we'll then be able to better analyze and implement performance improvements for the mobile app

@amyblais any follow up on this? App still seems to be unusable for the most part

**Comment by amyblais** [comment]
> > we'll then be able to better analyze and implement performance improvements for the mobile app
> 
> any follow up on this? App still seems to be unusable for the most part

Metrics were implemented in the latest release (July 16th), and analysis/investigation continues over the next upcoming months.

**Comment by vetash** [comment]
Any updates on that?

**Comment by aloknikhil** [comment]
This basically forces our users to uninstall and reinstall the entire app every few weeks. Any idea on what's going on?

**Comment by vetash** [comment]
Still absolutely unusable after several years

**Comment by ShiroDN** [comment]
Yeah, mobile apps are still utterly unusable. Even channel retention, which should help by cleaning up some very busy channels, hasn’t made a difference. Someone from mattermost mentioned that it should also clean up the mobile database, but we tried it during the 30-day trial, and it didn’t help. If it had worked, we’d already be on the Pro plan. But right now, my boss would throw me out the window if I brought up paying for mattermost while the mobile apps are still broken.

**Comment by vetash** [comment]
Web version on mobile works perfectly. I've created android app with  web view on https://creator.kodular.io,  it took about 5 minutes.  Unfortunately, that method does not support push notification, but at least it works without freezes.  :smiley:

**Comment by fussyrabbit** [comment]
@amyblais Hi! Is there any update on this issue? After 5 days, the mobile app is still utterly unusable.

**Comment by amyblais** [comment]
We have these Jira Epics open with several linked child issues to address mobile performance issues that you can follow:
 - https://mattermost.atlassian.net/browse/MM-58231
 - https://mattermost.atlassian.net/browse/MM-61313

**Comment by aleho** [comment]
@amyblais I really don't want to jump on the "Why's Mattermost Mobile not yet fixed" bandwagon, but these epics are about "low bandwith networks". Are the titles just inaccurate? I'm using the mobile app on a 400MBit WiFi / internet connection and after a few days it's so slow simply opening a channel takes half minute whereas the mobile browser is reacting instantly.

**Comment by vetash** [comment]
Seems like 10.11 and latest mobile 2.31.0 finally works fine. Even on old android device. 

I think devs should get into this issue for getting realize what exact changes make MM works fast.
